### PR TITLE
Bratseth/propagate node repo stats

### PR DESCRIPTION
**Pull request by bratseth**

### CODE REVIEW

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/controller/StatsResponse.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            Slime slime = new Slime();
+            Cursor root = slime.setObject();
+            Cursor zonesArray = root.setArray("zones");
+            for (ZoneId zone : controller.zoneRegistry().zones().all().ids()) {
+                NodeRepoStats stats = controller.serviceRegistry().configServer().nodeRepository().getStats(zone);
+                if (stats.applicationStats().isEmpty()) continue; // skip empty zones
+                Cursor zoneObject = zonesArray.addObject();
+                zoneObject.setString("id", zone.toString());
+                toSlime(stats.load(), zoneObject.setObject("load"));
+                toSlime(stats.load(), zoneObject.setObject("activeLoad"));
```
Comment (freva) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/controller/StatsResponse.java:35]:
```suggestion
                toSlime(stats.activeLoad(), zoneObject.setObject("activeLoad"));
```

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/controller/StatsResponse.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                Cursor applicationsArray = zoneObject.setArray("applications");
+                for (var applicationStats : stats.applicationStats())
+                    toSlime(applicationStats, applicationsArray.addObject());
+            }
+            return slime;
+        }
+        catch (Exception e) {
+            e.printStackTrace();
```
Comment (freva) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/controller/StatsResponse.java:43]: try/catch just for testing? Remove?

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/restapi/NodesV2ApiHandler.java [hunk 3]
```diff
@@ -452,6 +454,34 @@ private HttpResponse application(String idString, URI uri) {
...
     }
 
+    private HttpResponse stats() {
+        var stats = nodeRepository.computeStats();
+
+        Slime slime = new Slime();
+        Cursor root = slime.setObject();
+
+        toSlime(stats.load(), root.setObject("load"));
+        toSlime(stats.load(), root.setObject("activeLoad"));
```
Comment (freva) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/restapi/NodesV2ApiHandler.java:464]:
```suggestion
        toSlime(stats.activeLoad(), root.setObject("activeLoad"));
```

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/restapi/NodesV2ApiHandler.java [hunk 4]
```diff
@@ -452,6 +454,34 @@ private HttpResponse application(String idString, URI uri) {
...
+        Cursor applicationsObject = root.setArray("applications");
```
Comment (freva) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/restapi/NodesV2ApiHandler.java:465]: `applicationsArray`

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/controller/StatsResponse.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    public StatsResponse(Controller controller) {
+        super(toSlime(controller));
+    }
+
+    private static Slime toSlime(Controller controller) {
+        try {
+            Slime slime = new Slime();
+            Cursor root = slime.setObject();
+            Cursor zonesArray = root.setArray("zones");
+            for (ZoneId zone : controller.zoneRegistry().zones().all().ids()) {
```
Comment (freva) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/controller/StatsResponse.java:29]: reach
```suggestion
            for (ZoneId zone : controller.zoneRegistry().zones().reachable().ids()) {
```

**Comment by bratseth** [comment]
Thanks!

**Pull request merged**